src/pint/models/stand_alone_psr_binaries/BT_piecewise.py

```diff
@@ -0,0 +1,441 @@
+import astropy.constants as c
+import astropy.units as u
+import numpy as np
+import pint.toa
+from pint import GMsun, Tsun, ls
+from pint.models.stand_alone_psr_binaries.BT_model import BTmodel
+from .binary_generic import PSR_BINARY
+
+
+class BTpiecewise(BTmodel):
+    def __init__(self, axis_store_initial=None, t=None, input_params=None):
+        self.binary_name = "BT_piecewise"
+        super(BTpiecewise, self).__init__()
+        self.axis_store_initial = []
+        self.extended_group_range = []
+        self.d_binarydelay_d_par_funcs = [self.d_BTdelay_d_par]
+        if t is not None:
+            self._t = t
+        if input_params is not None:
+            if self.T0X is None:
+                self.update_input(input_params)
+            elif self.T0X is not None:
+                self.update_input()
+        self.binary_params = list(self.param_default_value.keys())
+        # self.param_aliases.update({"T0X": ["T0X"], "A1X": ["A1X"]})
+
+    def set_param_values(self, valDict=None):
+        # print(valDict)
+        super().set_param_values(valDict=valDict)
+        self.piecewise_parameter_loader(valDict=valDict)
+
+    def piecewise_parameter_loader(self, valDict=None):
```

Comment:
It's preferable if a function name is an imperative - `load_piecewise_parameters`, perhaps, or `setup_internal_structures` or something. This function should have a docstring so you can explain what it does and when to call it.

Reply:
Forgot to add the docstring with the update, still to do...